plugins/pulp_python/plugins/querysets.py

```diff
@@ -0,0 +1,39 @@
+import itertools
+
+from pulp.server.controllers import repository as repo_controller
+from pulp.server.db.querysets import QuerySetPreventCache
+
+
+class PythonPackageQuerySet(QuerySetPreventCache):
+    """
+    Custom querysets for Python packages.
+    """
+
+    def packages_in_repo(self, repo_id):
+        """
+        Query for all Python packages in a repository.
+
+        :param repo_id: Identifies the repository from which to retrieve packages
+        :type  repo_id: basestring
+        :return: QuerySet containing each package in the repo
+        :rtype:  mongoengine.queryset.QuerySet
+        """
+        unit_qs = repo_controller.get_unit_model_querysets(repo_id, self._document)
+        return itertools.chain(*unit_qs)
+
+    def packages_by_project(self, repo_id):
+        """
+        Query for all Python packages, organized by project.
+
+        :param repo_id: Identifies the repository from which to retrieve packages
+        :type  repo_id: basestring
+        :return: Dictionary containing packages in the repo, organized by project name
+        :rtype:  dict
+        """
+        packages = self.packages_in_repo(repo_id)
+        projects = {}
+
+        for pkg in packages:
```

Comment:
I did sleep on this and unrelated to whether or not we have a ``Project`` model, this could probably be written to return a generator of tuples. Something like ``(name, (list, of, packages))``. However, I'm happy spend the weekend thinking about what we want to do here.